### ISSUE TITLE
Site defaults to Chinese

### ISSUE THREAD

**Issue by sebbASF** [apache/echarts#15620]
### Version
Current

### Steps to reproduce
Browse to https://echarts.apache.org/

### What is expected?
Site should redirect  based on request header such as accept-language

### What is actually happening?
Unconditional redirect to Chinese site

**Comment by pissang** [comment]
We are using `navigator.userLanguage` to redirect. Do you mind giving us your value?

**Comment by sebbASF** [comment]
Those look like Javascript variables. It should not be necessary to enable JavaScript for the redirect to work.

User agents such as browsers send a header such as mine:

accept-language: en-GB,en;q=0.9

AFAIK, if the HTTP server and the page paths are set up correctly, it should redirect automatically without needing Javascript.

**Comment by sebbASF** [comment]
Here is some info on setting up the Apache server to handle different languages
https://httpd.apache.org/docs/2.4/content-negotiation.html

**Comment by Ovilia** [comment]
It seems to do with our caching settings. We are working on this. Thanks for reporting!

**Comment by pissang** [comment]
@sebbASF I changed redirection from 301 moved permanently to 302 found. It should work properly now. Sorry for the mistake.

**Comment by sebbASF** [comment]
It does not always work for curl:

$ curl -I -H "Accept-language: en-GB,en;q=0.9" echarts.apache.org
HTTP/1.1 302 Found
Server: Apache
Location: https://echarts.apache.org/zh/index.html

Then a few minutes later:

$ curl -I -H "Accept-language: en-GB,en;q=0.9" echarts.apache.org
HTTP/1.1 302 Found
Server: Apache
Location: https://echarts.apache.org/en/index.html

I'm not 100% sure it always works for browsers; will have to see.

**Comment by plainheart** [comment]
It seems the redirection will be cached if we request the same URL even if `Accept-Language` is changed.
It works well if we add a query parameter, for example, `echarts.apache.org?v=1`

**Comment by sebbASF** [comment]
Yes, but adding a parameter only works if it is changed from the previous one, and you can't expect users to do that.

The cache key needs to include accept-language.

**Comment by Ovilia** [comment]
@plainheart has probably made some updates about the cache policy. Waiting for him to give a comment.

**Comment by plainheart** [comment]
@sebbASF Hi, we've fixed the redirection issue. Here is the current behavior,

<img src="https://user-images.githubusercontent.com/26999792/133919101-4ae5cd2b-0bfd-4ff7-9889-6727a39af16d.png" width="500">
